squirrels/_auth.py

```
@@ -468,6 +411,298 @@ def get_user_from_token(self, token: str | None) -> tuple[RegisteredUser | None,
+            )
+
+        provider = next((p for p in self.auth_providers if p.name == provider_name), None)
+        if provider is None:
+            raise InvalidInputError(404, "auth_provider_not_found", f"Provider '{provider_name}' not found")
+
+        metadata: dict = self._get_provider_metadata(provider_name)
+
+        userinfo_endpoint = metadata.get("userinfo_endpoint")
+        if isinstance(userinfo_endpoint, str) and userinfo_endpoint:
+            try:
+                response = requests.get(
```

Ruff (0.14.13)

Probable use of `requests` call without timeout

(S113)